spec/unit/chef_server_artifact_spec.rb

```diff
@@ -53,12 +55,37 @@
 
   it "provides lock data as a Hash" do
     expected_data = {
-      "chef_server" => url,
-      "server_identifier" => cookbook_identifier,
+      "chef_server_artifact" => url,
+      "identifier" => cookbook_identifier,
     }
     expect(chef_server_artifact_location.lock_data).to eq(expected_data)
   end
 
+  it "returns a cached cookbook object" do
+    expect(MockCachedCookbook).to receive(:from_path).with(Pathname.new(expected_install_path))
+    chef_server_artifact_location.cached_cookbook
+  end
+
+  context "when using the default chef server HTTP client" do
```

Comment:
Should we have a test for making sure the thing provided in options wins if both are specified?

Reply:
done